.github/workflows/ci.yml

```diff
@@ -0,0 +1,153 @@
+name: CI
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - master
+      - develop
+      - "release/**"
+env:
+  APPLITOOLS_API_KEY: ${{ secrets.APPLITOOLS_API_KEY }}
+  APPLITOOLS_BATCH_ID: ${{ github.sha }}
+  APPLITOOLS_BATCH_NAME: ${{ github.head_ref }}
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: v12.20.0
+      - uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: |
+            **/node_modules
+            ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
```

Comment:
were you able to confirm that this cache works **across** feature branches? This is only part of caching we haven't solved yet. The other changes here are close to iterations we've had before

Reply:
I performed a cursory test that suggested it didn't work across feature branches. I cant be sure though and will keep an eye on it post merge.

The caching _definitely_  works within the branch and explicitly skips the dependency install step (see screenshot) this is functionality I never saw working before.

![CleanShot 2021-10-04 at 14 37 21@2x](https://user-images.githubusercontent.com/62259644/135928277-a73d1b08-bb3e-468e-9f18-e39886c779e3.png)

Reply:
Thanks @arnoldsandoval-okta [this is the same functionality we had in this iteration](https://github.com/okta/odyssey/pull/945). We also [later experimented with caching via the yarn cache](https://github.com/okta/odyssey/pull/1033) (rather than node_modules directories) while at the same time fixing CI to use the correct node version. What we really want is caching across feature branches because when the yarn.lock doesn't change we should be able to speed up CI by ~4 minutes.

Reply:
here are the docs that seem to describe the functionality we're looking for. I believe `develop` is setup as our default github branch as well but we don't have permissions to confirm in github ourselves.

https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key